x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/enrollment/tool/BootstrapPasswordAndEnrollmentTokenForInitialNode.java

```diff
@@ -0,0 +1,187 @@
+
+public class BootstrapPasswordAndEnrollmentTokenForInitialNode extends KeyStoreAwareCommand {
+    private static final String elasticUser = ElasticUser.NAME;
+    private KeyStoreWrapper keyStoreWrapper;
+    private final CheckedFunction<Environment, EnrollmentTokenGenerator, Exception> createEnrollmentTokenFunction;
+    private final Function<Environment, CommandLineHttpClient> clientFunction;
+    private final CheckedFunction<Environment, KeyStoreWrapper, Exception> keyStoreFunction;
+    private final  OptionSpec<String> keyStorePassword;
+    private final  OptionSpec<Void> docker;
+    final SecureRandom secureRandom = new SecureRandom();
+
+    // Package-private for testing
+    CommandLineHttpClient getClient(Environment env) {
+        return clientFunction.apply(env);
+    }
+    EnrollmentTokenGenerator getEnrollmentTokenGenerator(Environment env) throws Exception{
+        return createEnrollmentTokenFunction.apply(env);
+    }
+    KeyStoreWrapper getKeyStoreWrapper(Environment env) throws Exception { return keyStoreFunction.apply(env); }
+    OptionParser getParser() { return parser; }
+
+    BootstrapPasswordAndEnrollmentTokenForInitialNode() {
+        this(
+            environment -> new CommandLineHttpClient(environment),
+            environment -> KeyStoreWrapper.load(environment.configFile()),
+            environment -> new EnrollmentTokenGenerator(environment)
+        );
+    }
+
+    BootstrapPasswordAndEnrollmentTokenForInitialNode(Function<Environment, CommandLineHttpClient> clientFunction,
+                                                      CheckedFunction<Environment, KeyStoreWrapper, Exception> keyStoreFunction,
+                                                      CheckedFunction<Environment, EnrollmentTokenGenerator, Exception>
+                                                     createEnrollmentTokenFunction){
+        super("Set elastic password and generate enrollment token for initial node");
+        this.clientFunction = clientFunction;
+        this.keyStoreFunction = keyStoreFunction;
+        this.createEnrollmentTokenFunction = createEnrollmentTokenFunction;
+        parser.allowsUnrecognizedOptions();
+        keyStorePassword = parser.accepts("keystore-password", "keystore password").withRequiredArg();
+        docker = parser.accepts("docker", "determine that we are running in docker");
+    }
+
+    public static void main(String[] args) throws Exception {
+        exit(new BootstrapPasswordAndEnrollmentTokenForInitialNode().main(args, Terminal.DEFAULT));
+    }
+
+    @Override
+    protected void execute(Terminal terminal, OptionSet options, Environment env) throws Exception {
+        if (!options.has(keyStorePassword)) {
+            throw new UserException(ExitCodes.USAGE, null);
+        }
+
+        final SecureString keystorePassword = new SecureString(keyStorePassword.values(options).get(0));
+        SecureString password;
```

Comment:
I think that if you first generate tokens and then set the password of the elastic user, it would allow you to simplify your code as you don't need to keep track of password here and first use boostrapPassword. You can keep using bootstrapPassword for everyhing and just print the autogenerated password of the user in the end